### ISSUE TITLE
The use of the Data class 

### ISSUE THREAD

**Issue by almarklein** [vispy/vispy#62]
Related to my previous email, I have some comments about the data class.

Firstly, it should probably be called "VertexData" to avoid confusion.

Secondly, I don't really get what problem it solves. It seems like it is basically syntactic sugar, so people can instantiate one class instead of having to create a numpy array and vertexbuffer. Can someone show an example code with and without the data class?

Basically, I wonder whether this is worth an extra class. Or can we bring this concept further in some way, perhaps by offering this functionality as a VertexBuffer that stores a local copy?
